src/algebra/group_action_hom.lean

```diff
@@ -162,6 +175,22 @@ protected def id : A →+[M] A :=
 
 variables {M A B C}
 
+instance : has_zero (A →+[M] B) :=
+⟨{ map_smul' := by simp,
+   .. (0 : A →+ B) }⟩
+
+instance : has_one (A →+[M] A) := ⟨distrib_mul_action_hom.id M⟩
+
+@[simp] lemma coe_zero : ((0 : A →+[M] B) : A → B) = 0 := rfl
+
+@[simp] lemma coe_one : ((1 : A →+[M] A) : A → A) = id := rfl
```

Comment:
I'm a little surprised you added these - they're certainly worth having eventually, but I'm surprised they're useful to you without also `has_add` (elementwise) and `has_mul` (composition), along with the associated monoid structures.

Reply:
You're quite right that I'm not using them but my view is that there's not much point in adding a `one` or `zero` if I don't characterise them.

Reply:
Oh I think you mean you're surprised that I added instances of `has_zero`, `has_one` rather than the `coe_zero`, `coe_one` lemmas given that I added the instances.

Again I don't need them but I figured that while I was here I might as well remove the `@[nolint has_inhabited_instance]` by adding zero as a default and then I figured if I was adding zero, why not add one!

Reply:
Yes exactly - this seemed orthogonal to both the rest of this PR and your original, so I thought I'd check why it was here. Satisfying the linter was also my best guess.